### ISSUE TITLE
Why does ActionModeHelper.onClick never return true?

### ISSUE THREAD

**Issue by ankitbatra11** [mikepenz/FastAdapter#382]
ActionModeHelper.onClick never returns true which means below code in onPreClickListener (as given in multi select sample):

```
Boolean result = actionModeHelper.onClick(item);
Timber.d("onPreClick: result=%s", result != null ? result.toString() : "null");
return result != null && result;
```

would not consume the event if this item was selected or deselected and eventually trigger onClick of an item (if onClickListener is set).

**Comment by mikepenz** [comment]
@ankitbatra11 sorry for the late answer. 
The reason for this is, that the selecting of an item will happen in the internal parts of the `FastAdapter` and not by the `ActionModeHelper` this results in the need that the `ActionModeHelper` can't `consume` an event, as otherwise the item wouldn't be selected / deselected. 

This changed a bit since v1, that's why the sample still shows wrong information. 

Checking in your `onClick` if the `ActionModeHelper` is currently active should already handle most click cases. 

```java
if(actionModeHelper.isActive()) {
    return false;
}
```

In addition I have to update the `ActionModeHelper` and consume the event for the last deselection as in this case the `ActionModeHelper` would no longer be active (already resulting in a triggered click) which is not good.

**Comment by mikepenz** [comment]
This will be fixed in the next rlease by: https://github.com/mikepenz/FastAdapter/commit/2175fedd4e7664e72fc4ea7626c94a92fcc9100e

**Comment by mikepenz** [comment]
Fixed in v2.5.2
